aten/src/ATen/native/mps/operations/RangeFactories.mm

```diff
@@ -53,8 +53,57 @@
     }
     bool is_contiguous = result.is_contiguous();
     Tensor r = !is_contiguous ? at::empty_like(result, LEGACY_CONTIGUOUS_MEMORY_FORMAT) : result;
+    using namespace mps;
+    struct CachedGraph : public MPSCachedGraph {
+      CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+      MPSGraphTensor *startTensor = nil;
+      MPSGraphTensor *multiplyTensor = nil;
+      MPSGraphTensor *outputTensor = nil;
+    };
+    auto cache_ = MPSGraphCache::getInstance();
+    auto stream = getCurrentMPSStream();
+    auto mpsDataType = getMPSDataType(result.scalar_type());
+    @autoreleasepool {
+      string key = "arange_mps_out:" + getTensorsStringKey({result}) + ":" + to_string(size);
```

Comment:
This doesn't need to be parametrized by the start/step?

Reply:
Not really, as `start` and `step` are added as graph parameters

Reply:
Ho ok!